crates/forge_infra/src/env.rs

```diff
@@ -127,10 +128,101 @@ impl ForgeEnvironmentService {
             retry_config,
         }
     }
+
+    /// Load all `.env` files with priority to lower (closer) files.
+    fn load_all(cwd: &Path) -> Option<()> {
+        let mut paths = vec![];
+        let mut current = PathBuf::new();
+        
+        for component in cwd.components() {
+            current.push(component);
+            paths.push(current.clone());
+        }
+        
+        paths.reverse();
+        
+        for path in paths {
+            let env_file = path.join(".env");
+            if env_file.is_file() {
+                dotenv::from_path(&env_file).ok();
+            }
+        }
+        
+        Some(())
+    }
 }
 
 impl forge_domain::EnvironmentService for ForgeEnvironmentService {
     fn get_environment(&self) -> Environment {
         self.get()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+    use std::env;
+    use tempfile::tempdir;
+
+    fn write_env_file(dir: &Path, content: &str) {
+        let env_path = dir.join(".env");
+        fs::write(&env_path, content).unwrap();
+    }
+
+    #[test]
+    fn test_load_all_single_env() {
+        let dir = tempdir().unwrap();
+        write_env_file(dir.path(), "TEST_KEY1=VALUE1");
+
+        ForgeEnvironmentService::load_all(dir.path());
+
+        assert_eq!(env::var("TEST_KEY1").unwrap(), "VALUE1");
+    }
+
+    #[test]
+    fn test_load_all_nested_envs_override() {
+        let root = tempdir().unwrap();
+        let subdir = root.path().join("subdir");
+        fs::create_dir(&subdir).unwrap();
+
+        write_env_file(root.path(), "TEST_KEY2=ROOT");
+        write_env_file(&subdir, "TEST_KEY2=SUB");
+
+        ForgeEnvironmentService::load_all(&subdir);
+
+        assert_eq!(env::var("TEST_KEY2").unwrap(), "SUB");
```

Comment:
```suggestion
        let envs = test_env([
          "/a/b/c", "A=1",
          "/a/b", "A=1"
        ]);

        assert_eq!(env::var("A").unwrap(), "1");
```

Make the tests easier to understand.